src/chapters/loops/for-loops.rst

```diff
@@ -0,0 +1,90 @@
+**for** Loops
+=============
+
+.. index::
+   single: loop; for
+   single: iteration; definite
+
+.. index:: ! for loop
+
+The ``for`` loop is the first JavaScript tool for iteration that we will explore. A **for loop** allows us to carry out **definite iteration**. Definite iteration is the process of repeating a specific task with a specific data set. In other words, when a ``for`` loop begins it can usually be determined exactly how many times it will execute: once for each item in the data set.
+
+**for** Loop Syntax
+-------------------
+
+.. index::
+   single: for loop; syntax
+
+We have already seen the basic syntax of a ``for`` loop.
+
+.. sourcecode:: js
+
+   for (let i = 0; i < 51; i++) {
+      console.log(i);
+   }
+
+This program prints the integers 0 through 50. In the language of definite iteration, we can say that the operation that this loop carries out printing a value to the console, and its data set is the collection of integers 0...50.
+
+Let's break down this syntax piece by piece, so we can begin to understand how ``for`` loops are structured.
+
+.. index::
+   single: for loop; variable
+   single: for loop; initial expression
+   single: for loop; condition
+   single: for loop; update expression
+
+A ``for`` loop always contains the following components:
+
+::
+
+   for (initial expression; loop condition; update expression) {
+      loop body
+   }
+
+Notice that in the first line, within parentheses, the components **initial expression**, **loop condition**, and **update expression** are separated by semicolons. Let's look at these components in detail.
+
+- The statement ``let i = 0`` is executed exactly once, at the *beginning* of loop execution. The variable ``i`` is the **loop variable**. 
+- The boolean expression ``i < 51`` is the **loop condition**. This condition is evaluated before each loop iteration, or repetition. If the condition is ``true`` then the loop executes again. If the condition is ``false`` then the loop ceases execution and the program continues to execute the code below the loop. In the example above, an equivalent condition would be ``i <= 50``. It is common to use ``<`` when writing loop conditions, so that the last value used in the loop is *one less than* then right-hand side of the ``<`` symbol.
```

Comment:
Grammar and typos. How about, “The boolean expression ``i < 51`` is the **loop condition**. This condition is evaluated before each loop iteration. If the condition is ``true`` then the loop executes. If the condition is ``false`` then the loop ceases, and the program continues with the code that follows the loop. In the example above, an equivalent condition would be ``i <= 50``. It is common to use ``<`` when writing loop conditions, so that the last value used in the loop is *one less than* the right-hand side of the ``<`` symbol.”

Reply:
OK.